ASUWebPlatforms/webspark-ci
# WS2-1773: Fix FontAwesome composer problem and add FontAwesome Icon Picker Widget support for ASUAwesome

### Description

Sprint 45 includes updating the Drupal FontAwesome module to the dev branch. However, this causes Composer to throw an error if the code changes are accepted. This is because there is a mismatch with the patch we are applying (#3219579: Hidding font awesome additional settings). The Dev branch of the module fixes this issue, which makes the patch outdated. So, we need to remove the patch and adjust how we handle hiding the additional settings. The Dev branch handles this situation by creating a new permission. We just need to make sure that permission is not enabled for anyone except Administrators.

The other part of this ticket is that we need to get the ASUAwesome icons to be included in the FontAwesome Icon Picker Widget. We do this by slightly changing the approach of how we are including ASUAwesome. I was using a custom prefix (`fasu`), but the icon picker doesn’t recognize that. So, I found that I can use the base `fas` prefix, and include our icons as part of the “solid” icon library. 

The way to make our icons available is to include a metadata YAML file which is made available to the FontAwesome Icon Picker Widget module. To accomplish this, I ended up employing a Composer post-update function that operates similarly to how we handle patches. It takes our FontAwesome library and combines it with the ASUAwesome library. This is then read by the icon picker module and loaded. I have the ASUAwesome icons displaying first, so that people can find them easily.

Lastly, I realized that my initial approach in how I was including FontAwesome was a little flawed, because we weren’t including the regular icons. After looking into it further, it makes the most sense to use the `all.js` version, after all. It doesn’t include the pro icons, just the free ones. So, I have adjusted things to include the free “regular” icon set as well. This simplifies things, and makes everything work.
QA Testing:

To test this ticket, we need to do the following:

- Check a page that has FontAwesome icons on it. I recommend viewing the Degree Details page (`/bachelors-degrees/majorinfo/ASACOCBS/undergrad/false/37`). All icons should appear without any issues.
- Upgrading to FA6 means that some icons are no longer supported in the “brands” section. On the `/card-arrangement` page, you will see in the “Vertical Arrangement” section, one of the cards has a FontAwesome icon that is broken. This is because it is no longer part of the “brands” library. This is a good opportunity to test whether the ASUAwesome icons are included in the icon picker. So, click on “Layout” and scroll down to that section. Click on the pencil icon in the “Vertical Arrangement” block and then “configure” to bring up the editor. In the drawer that opens from the right, you can click on “cards” and then scroll down to the icon picker widget. In the widget, if you click the downward arrow, you should see icons on the first page of icons that belong exclusively to ASUAwesome. For example, you should see an “A Mountain” icon, a couple of hand pitchfork icons, some shapes, etc. These are all part of the ASUAwesome custom icon library. [You can see more about this library here](https://docs.google.com/spreadsheets/d/1FOx9yVlhv5GZuBhWgeoXFuf7kmo4-IIou2r2BUmkdW4/edit?usp=sharing).
-- If you are logged in as a user who is not a site administrator, you should not be able to see the “Additional Font Awesome Settings” section just below the icon picker.
-- If you click through several “pages” of the icons, you should not see any missing or blank spots. All icons should be present, and there should not be duplicates of any icons.
- Please note, this is a “breaking change” for ASUAwesome, because we are changing the prefix from `fasu` to `fas`. I don’t think many people are using ASUAwesome, because it wasn’t included in the icon picker, but, we should let people know that if they have manually added any ASUAwesome icons, they will need to adjust accordingly.

### Links

- [JIRA ticket](https://asudev.jira.com/browse/WS2-1773)

### Checklist


### Verified in browsers 


### Screenshots



Note: Sections that are not applicable for this issue can be removed.
